src/Helpers/LeanCode.UserIdExtractors/Extractors/RawTypedUserIdExtractor.cs

```diff
@@ -0,0 +1,36 @@
+using System.Globalization;
+using System.Security.Claims;
+using LeanCode.DomainModels.Ids;
+
+namespace LeanCode.UserIdExtractors.Extractors;
+
+internal class RawTypedUserIdExtractor<TBacking, TId> : StringUserIdExtractor, IUserIdExtractor<TId>
+    where TBacking : struct
+    where TId : struct, IRawTypedId<TBacking, TId>
+{
+    private static readonly Dictionary<Type, Func<string, object>> Parsers = new Dictionary<Type, Func<string, object>>
```

Comment:
Let's go with [`ReadOnlyDictionary`](https://learn.microsoft.com/en-us/dotnet/api/system.collections.objectmodel.readonlydictionary-2?view=net-8.0).

On the other hand, I would say that we should add `Parse` to the `IRawTypedId` so that we don't need the dictionary at all.

Reply:
I'd say we should go a step further and use `FrozenDictionary` for such cases.

Reply:
Ah! That's the one I had in mind TBH, but mistook the name.